tchannel-core/src/main/java/com/uber/tchannel/utils/TChannelUtilities.java

```diff
@@ -38,26 +39,57 @@
 
     public static final ByteBuf emptyByteBuf = Unpooled.EMPTY_BUFFER;
 
+    public static int scoreAddr(NetworkInterface iface, InetAddress addr) {
+        int score = 0;
+
+        if (addr instanceof Inet4Address) {
+            score += 300;
+        }
+
+        try {
+            if (!iface.isLoopback() && !addr.isLoopbackAddress()) {
```

Comment:
@venkat1109 check my example in the description, this changes it to prefer 10 over 179 and (less obviously) to prefer eth0 over eth1. Link-local addresses are scored lower.